samples/emoji-search/values/src/commonMain/kotlin/example/values/IntervalList.kt

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.values
+
+import androidx.compose.runtime.collection.MutableVector
+import androidx.compose.runtime.collection.mutableVectorOf
+import kotlinx.serialization.Serializable
+
+/**
+ * The list consisting of multiple intervals.
+ *
+ * It is useful when you want to define your own dsl similar to the one used by LazyColumn where
+ * list items can be defined via multiple item/items calls.
+ *
+ * This interface is read only, in order to create a list you need to use [MutableIntervalList].
+ *
+ * @param T type of values each interval contains in [Interval.value].
+ */
+sealed interface IntervalList<T> {
```

Comment:
I wonder if there’s a better name for this thing. Aggregate or Concatenated both come to mind.

Reply:
This class is a copy-paste of `androidx.compose.foundation.lazy.layout.IntervalList` so I'll probably keep everything as it's what `LazyColumn` in Compose UI is using under the hood too. I thought I changed the license of this file to Android's, but I think the `.idea/copyright/profiles_settings.xml` keeps on overwriting it as soon as I leave the file…